### PR TITLE
Apply repo-review suggestions

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
https://learn.scientific-python.org/development/guides/repo-review/?repo=zarr-developers%2Fnumcodecs&branch=main

TODO:

- [ ] Unit tests and/or doctests in docstrings
- [ ] Tests pass locally
- [ ] Docstrings and API docs for any new/modified user-facing classes and functions
- [ ] Changes documented in docs/release.rst
- [ ] Docs build locally
- [x] GitHub Actions CI passes
- [x] Test coverage to 100% (Codecov passes)

### CODE REVIEW

**Comment by joshmoore** [comment]
Thanks, @DimitriPapadopoulos, but would you also be willing to apply this to the `v3` branch? v2 (main) is entering maintenance mode: https://zarr.dev/blog/zarr-python-v3-update/

**Comment by DimitriPapadopoulos** [comment]
There is no `v3` branch in `numcodecs`, is there? This is `numcodecs`, not `zarr-python`.

**Comment by joshmoore** [comment]
:facepalm: Apologies, @DimitriPapadopoulos. You are of course correct. Sorry for the noise.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+exclude: ^fixture/
```
Comment (dstansby) [inline .pre-commit-config.yaml:1]: Could you add a comment explaining why this is ignored?

File: adhoc/blosc_memleak_check.py [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-import numcodecs
 import numpy as np
 from numpy.testing import assert_array_equal
 
+import numcodecs
 
 codec = numcodecs.Blosc()
 data = np.arange(int(sys.argv[1]))
-for i in range(int(sys.argv[2])):
+for _i in range(int(sys.argv[2])):
```
Comment (dstansby) [inline adhoc/blosc_memleak_check.py:10]:
```suggestion
for _ in range(int(sys.argv[2])):
```
Any reason not to just go with a single underscore?

**Review by dstansby — APPROVED** [review]

File: adhoc/blosc_memleak_check.py (hunk 2 above)
Comment (DimitriPapadopoulos) [reply adhoc/blosc_memleak_check.py]: No, I was trying to stay as close as possible to the original code, but in this case it doesn't make sense. Changed to `_`.

**Pull request merged**